src/contracts/AccessControlShares.sol

```diff
@@ -0,0 +1,36 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.23;
+
+import {MultiLP} from "./MultiLP.sol";
+
+abstract contract AccessControlShares is MultiLP {
+    uint256 public totalSupplyCap;
+    mapping(address lp => uint256 shares) public liquidityProviderCaps;
+
+    uint256[50] private _gap;
```

Comment:
Our gaps should take into account the number of other storage slots used. So this should be 48, if I have my math right.